internal/services/azurestackhci/stack_hci_virtual_hard_disk_resource.go

```diff
@@ -0,0 +1,321 @@
+			ValidateFunc: validation.IntAtLeast(1),
+		},
+
+		"dynamic_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+
+		"hyperv_generation": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(virtualharddisks.HyperVGenerationVOne),
+				string(virtualharddisks.HyperVGenerationVTwo),
+			}, false),
+		},
+
+		"logical_sector_in_bytes": {
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntAtLeast(1),
+		},
+
+		"physical_sector_in_bytes": {
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntAtLeast(1),
+		},
+
+		"storage_path_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: storagecontainers.ValidateStorageContainerID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (StackHCIVirtualHardDiskResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCIVirtualHardDiskResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.VirtualHardDisks
+
+			var config StackHCIVirtualHardDiskResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := virtualharddisks.NewVirtualHardDiskID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := virtualharddisks.VirtualHardDisks{
+				Name:     pointer.To(config.Name),
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				ExtendedLocation: &virtualharddisks.ExtendedLocation{
+					Name: pointer.To(config.CustomLocationId),
+					Type: pointer.To(virtualharddisks.ExtendedLocationTypesCustomLocation),
+				},
+				Properties: &virtualharddisks.VirtualHardDiskProperties{
+					Dynamic: pointer.To(config.DynamicEnabled),
+				},
+			}
+
+			if config.BlockSizeInBytes != 0 {
+				payload.Properties.BlockSizeBytes = pointer.To(config.BlockSizeInBytes)
+			}
+
+			if config.StoragePathId != "" {
+				payload.Properties.ContainerId = pointer.To(config.StoragePathId)
+			}
+
+			if config.DiskFileFormat != "" {
+				payload.Properties.DiskFileFormat = pointer.To(virtualharddisks.DiskFileFormat(config.DiskFileFormat))
+			}
+
+			if config.DiskSizeInGB != 0 {
```

Comment:
This is required, so can be embedded to the definition of the struct?

Reply:
fixed